src/windows/FileHashProxy.js

```diff
@@ -0,0 +1,232 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+*/
+
+/* global Windows, WinJS, MSApp */
+
+var getFileFromPathAsync = Windows.Storage.StorageFile.getFileFromPathAsync;
+
+
+function cordovaPathToNative(path) {
+    // turn / into \\
+    var cleanPath = path.replace(/\//g, '\\');
+    // turn  \\ into \
+    cleanPath = cleanPath.replace(/\\+/g, '\\');
```

Comment:
This looks dodgy. I think the second param should should be  '\' (not the double, right? Otherwise line 29 is dodgy)

With the regex, would that not replace '\\\\' with '\' ? I would've expected '/(\\)+/' maybe (or whatever the grouping brackets are)

Reply:
I dunno, this came from the cordova file plugin.

Reply:
So what's happening here, is that all the cordova paths come through with forward slashes. So it's trying to ensure that those get replaced with backslashes (which Windows uses), and then ensures that all instances where there are one or more backslashes get replaced with a double backslash.

`/\\+/g` is matching an individual backslash, globally. `\` is an escape character, so to actually use it we need to escape it with itself :)